### PR TITLE
refactor: using exclusive or outputs features for Token classifier

### PR DESCRIPTION
**Pull request by frascuchon**
This PR splits  `labels` field in `TokenClassification` head in `labels` (tag labels)  and `entities` (span labels). Consequently, multiple output can be defined in pipelines, returning a list of names, instead of a single string.

### CODE REVIEW

File: tests/text/modules/heads/test_token_classification.py [hunk 1]
```diff
@@ -60,6 +60,8 @@ def trainer_dict() -> Dict:
 def test_train(pipeline_dict, training_data_source, trainer_dict, tmp_path):
     pipeline = Pipeline.from_config(pipeline_dict)
 
+    assert pipeline.output == ["entities", "labels"]
```
Comment (dcfidalgo) [inline tests/text/modules/heads/test_token_classification.py:63]: This is a bit confusing. Right now the `pipeline.output` are the optional arguments of the `featurize` method of the head. But the `pipeline.predict` does not return a `labels`, but a `tags` ... maybe we should somehow enforce a consistency between the two things, or better define what `pipeline.output` actually means.

**Review by dcfidalgo — COMMENTED** [review]
Nice, much cleaner this way! 
I have a slight preference for 'tags' instead of 'labels' in the `featurize` method, but no strong opinion.
I just left a couple of inline comments.

File: tests/text/modules/heads/test_token_classification.py (hunk 1 above)
Comment (frascuchon) [reply tests/text/modules/heads/test_token_classification.py]: For now, I've updated the `featurize` method, renaming `labels` to `tags`.   

We must define a data model for task input/outputs that will keep this logic, reducing this kind of frictions/confussions.

File: src/biome/text/dataset.py [hunk 2]
```diff
@@ -312,7 +312,7 @@ def to_instances(self, pipeline: "Pipeline", lazy=True, num_proc: Optional[int]
     @staticmethod
     def _create_and_pickle_instances(row, input_columns, featurize, instances_col_name):
         """Helper function to be used together with the `datasets.Dataset.map` method"""
-        instance = featurize(**{key: row[key] for key in input_columns})
+        instance = featurize(**{key: row.get(key) for key in input_columns})
```
[the remark this answers is not in the document]
Comment (frascuchon) [reply src/biome/text/dataset.py]: Again, include data model for input and outputs we can easily handle this logic. 

Anyway, I've made some changes for take into account optional/non optionals inputs.

**Review by dcfidalgo — APPROVED** [review]
Nice, LGTM!

**Pull request merged**